src/charm.py

```diff
@@ -1119,6 +1094,8 @@ def is_primary(self) -> bool:
     @property
     def is_tls_enabled(self) -> bool:
         """Return whether TLS is enabled."""
+        if not self.model.get_relation(PEER):
+            return False
```

Comment:
Used in the charm's init to set the monitoring services, which fails with the current implementation when the charm initially bootstraps.

Reply:
Q: Wasn't it enough to prevent this in the `(get|set)_secret()` methods?
I came across the same issue on VM, thus I've added the same check to `(get|set)_secret()` if I recall right. I'm wondering shouldn't we rather add it to those functions instead?

I mean theoretically critical code should only be executeable from those two functions -- and then that should be the "central" source for this safety measure.

Reply:
Discussed with @juditnovak. The check is in the Mongo charms. I would prefer to keep this check outside of the secrets methods, as it is a edge case that shouldn't apply to most calls of those methods.